pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,282 @@
+                If float, should be between 0.0 and 1.0 and
+                represent the fraction of parameters to prune.
+                If int, it represents the absolute number
+                of parameters to prune.
+                If Callable, the function will be called on every epoch.
+
+            prune_on_epoch_end: bool flag determines call or not
+                to call pruning_fn on epoch end.
+
+            prune_on_fit_end: bool flag determines call or not
+                to call pruning_fn on fit end.
+
+            make_pruning_permanent: if True then all
+                reparametrization pre-hooks and tensors with mask
+                will be removed on fit end.
+
+            use_lottery_ticket_hypothesis: Wether to use algorithm describes in
+                "The lottery ticket hypothesis" (https://arxiv.org/pdf/1803.03635.pdf)
+
+            pruning_dim: if you are using structured pruning method you need
+                to specify dimension.
+
+            pruning_norm: if you are using ln_structured you need to specify norm.
+        """
+
+        self.use_global_unstructured = use_global_unstructured
+        self.parameters_to_prune = parameters_to_prune
+        self.prune_on_epoch_end = prune_on_epoch_end
+        self.prune_on_fit_end = prune_on_fit_end
+        self.use_lottery_ticket_hypothesis = use_lottery_ticket_hypothesis
+        self.parameter_names = parameter_names or self.PARAMETER_NAMES
+
+        for param_name in self.parameter_names:
+            if param_name not in self.PARAMETER_NAMES:
+                raise MisconfigurationException(
+                    f"The provided parameter_names {param_name} isn't in {self.PARAMETER_NAMES} "
+                )
+
+        if isinstance(pruning_fn, str):
+            if pruning_fn not in _PYTORCH_PRUNING_FUNCTIONS.keys():
+                raise MisconfigurationException(
+                    f"The provided pruning_fn {pruning_fn} isn't available with "
+                    f"PyTorch build-in {_PYTORCH_PRUNING_FUNCTIONS.keys()} "
+                )
+            if "unstructured" not in pruning_fn:
+                if pruning_dim is None:
+                    raise MisconfigurationException(
+                        "When requesting `structured` pruning, the `pruning_dim` should be provided."
+                    )
+                if pruning_fn == "ln_structured":
+                    if pruning_norm is None:
+                        raise MisconfigurationException(
+                            "When requesting `ln_structured` pruning, the `pruning_norm` should be provided."
+                        )
+
+                    pruning_fn = self.create_pruning_fn(pruning_fn, dim=pruning_dim, n=pruning_norm)
+            else:
+                pruning_fn = self.create_pruning_fn(pruning_fn)
+
+        self.pruning_fn = pruning_fn
+
+        if not (prune_on_epoch_end or prune_on_fit_end):
+            rank_zero_warn(
```

Comment:
Should this only be a warning? 
Should it not be `MisconfigurationException` as it does not make sense to have this callback in your trainer if it is never used?

Reply:
Done !